src/ripple/basics/promises.h

```diff
@@ -33,8 +34,10 @@
 
 namespace ripple {
 
+#define TO_STRING0(x) #x
+#define TO_STRING(x) TO_STRING0(x)
 #define ASSERT_OP(op, x, y) \
-    if (x op y) { std::cerr << #x " == " << x << " " #op " " << y << " == " #y; }
+    if (x op y) { std::cerr << __FILE__ ":" TO_STRING(__LINE__) ": " #x " == " << x << " " #op " " << y << " == " #y << std::endl; std::abort(); }
 #define ASSERT_EQ(x, y) ASSERT_OP(!=, x, y)
 #define ASSERT_NE(x, y) ASSERT_OP(==, x, y)
```

Comment:
Having `ASSERT_EQ` use the `!=` operator is probably going to cause some confusion and maintainability issues down the line. Maybe you've already tried this and it didn't work, but if not, consider:
```
#define TO_STRING0(x) #x
#define TO_STRING(x) TO_STRING0(x)
#define ASSERT_OP(op, x, y) \
    if (!(x op y)) { std::cerr << __FILE__ ":" TO_STRING(__LINE__) ": " #x " == " << x << " " #op " " << y << " == " #y << std::endl; std::abort(); }
#define ASSERT_EQ(x, y) ASSERT_OP(==, x, y)
#define ASSERT_NE(x, y) ASSERT_OP(!=, x, y)
```